File: minecraft/src/main/java/com/spider/minecraft/protocol/Protocol.java
package com.spider.minecraft.protocol;

/**
 * Spider 标准协议信令类型。
 *
 * <p>完全对齐 Spider 经典客户端 /shared/protocol.py 中的常量定义。
 * SpiderMinecraft 的 TCP 直连端口发出的数据包在结构上与 Spider 客户端完全一致，
 * 因此可以直接对接 Spider 官方服务端程序。
 */
public final class Protocol {

    private Protocol() {}

    // ===== 认证信令 =====
    /** 客户端→服务器：注册新身份（首次使用） */
    public static final String REGISTER = "REGISTER";
    /** 客户端→服务器：登录已有身份 */
    public static final String LOGIN = "LOGIN";
    /** 服务器→客户端：注册/登录成功 */
    public static final String AUTH_OK = "AUTH_OK";
    /** 服务器→客户端：注册/登录失败 */
    public static final String AUTH_FAIL = "AUTH_FAIL";

    // ===== 消息信令 =====
    /** 客户端→服务器：发送消息（加密信封） */
    public static final String SEND_MSG = "SEND_MSG";
    /** 服务器→客户端：投递消息给接收方 */
    public static final String RECV_MSG = "RECV_MSG";
    /** 服务器→发送方：消息已送达对方 */
    public static final String DELIVERY_RECEIPT = "DELIVERY_RECEIPT";
    /** 接收方→发送方：消息已读 */
    public static final String READ_RECEIPT = "READ_RECEIPT";

    // ===== 离线信令 =====
    /** 服务器→客户端：离线消息队列推送 */
    public static final String OFFLINE_QUEUE = "OFFLINE_QUEUE";
    /** 客户端→服务器：请求离线消息 */
    public static final String REQUEST_OFFLINE = "REQUEST_OFFLINE";

    // ===== 安全信令 =====
    /** 客户端→服务器：本机已被胁迫，通知所有联系人 */
    public static final String COMPROMISED = "COMPROMISED";
    /** 服务器→客户端：胁迫通知确认 */
    public static final String COMPROMISED_ACK = "COMPROMISED_ACK";

    // ===== 群组信令 =====
    /** 创建群组 */
    public static final String GROUP_CREATE = "GROUP_CREATE";
    /** 加入群组 */
    public static final String GROUP_JOIN = "GROUP_JOIN";
    /** 离开群组 */
    public static final String GROUP_LEAVE = "GROUP_LEAVE";
    /** 群组消息 */
    public static final String GROUP_MSG = "GROUP_MSG";
    /** 群组信息 */
    public static final String GROUP_INFO = "GROUP_INFO";
    /** 群组列表 */
    public static final String GROUP_LIST = "GROUP_LIST";

    // ===== 保活 =====
    /** 心跳 */
    public static final String HEARTBEAT = "HEARTBEAT";
    /** 心跳响应 */
    public static final String HEARTBEAT_ACK = "HEARTBEAT_ACK";

    // ===== 文件传输 =====
    /** 文件传输请求 */
    public static final String FILE_SEND = "FILE_SEND";
    /** 文件数据块 */
    public static final String FILE_CHUNK = "FILE_CHUNK";
    /** 文件传输完成 */
    public static final String FILE_COMPLETE = "FILE_COMPLETE";

    // ===== 公钥查询 =====
    /** 查询用户公钥 */
    public static final String GET_PUBKEY = "GET_PUBKEY";
    /** 公钥查询结果 */
    public static final String PUBKEY_RESULT = "PUBKEY_RESULT";

    // ===== 通用 =====
    /** 错误 */
    public static final String ERROR = "ERROR";
    /** 断开连接 */
    public static final String BYE = "BYE";
}

File: android/app/src/main/java/com/spider/android/network/Protocol.kt
package com.spider.android.network

/**
 * Spider 协议常量 — 与 Python 服务端 shared/protocol.py 完全对齐。
 */
object Protocol {
    // 协议版本（整数，与服务端 PROTOCOL_VERSION = 3 对齐）
    const val PROTOCOL_VERSION = 3

    // ===== 客户端 → 服务端 =====
    const val REGISTER = "REGISTER"
    const val LOGIN = "LOGIN"
    const val SEND_MSG = "SEND_MSG"
    const val QUERY_PUBKEY = "QUERY_PUBKEY"
    const val STORE_PUBKEY = "STORE_PUBKEY"
    const val COMPROMISED = "COMPROMISED"
    const val STORE_DEADMAN_MSG = "STORE_DEADMAN_MSG"
    const val DEADMAN_ACK = "DEADMAN_ACK"
    const val ADMIN_AUTH = "ADMIN_AUTH"
    const val ADMIN_CMD = "ADMIN_CMD"
    const val SEARCH_CONTACTS = "SEARCH_CONTACTS"
    const val LOOKUP_USER = "LOOKUP_USER"
    const val PING = "PING"
    const val PONG = "PONG"
    const val DELIVERY_RECEIPT = "DELIVERY_RECEIPT"
    const val READ_RECEIPT = "READ_RECEIPT"

    // ===== 服务端 → 客户端 =====
    const val LOGIN_OK = "LOGIN_OK"
    const val LOGIN_FAIL = "LOGIN_FAIL"
    const val REGISTER_OK = "REGISTER_OK"
    const val RECV_MSG = "RECV_MSG"
    const val OFFLINE_QUEUE = "OFFLINE_QUEUE"
    const val SEND_OK = "SEND_OK"
    const val READ_RECEIPT_DISABLED = "READ_RECEIPT_DISABLED"
    const val PUBKEY_RESULT = "PUBKEY_RESULT"
    const val COMPROMISED_ACK = "COMPROMISED_ACK"
    const val ADMIN_AUTH_OK = "ADMIN_AUTH_OK"
    const val ADMIN_AUTH_FAIL = "ADMIN_AUTH_FAIL"
    const val CMD_RESULT = "CMD_RESULT"
    const val BROADCAST = "BROADCAST"
    const val RATE_LIMITED = "RATE_LIMITED"
    const val ERROR = "ERROR"
    const val LOOKUP_USER_RESULT = "LOOKUP_USER_RESULT"

    // ===== 群组 =====
    const val CREATE_GROUP = "CREATE_GROUP"
    const val JOIN_GROUP = "JOIN_GROUP"
    const val LEAVE_GROUP = "LEAVE_GROUP"
    const val GROUP_ADD_MEMBER = "GROUP_ADD_MEMBER"
    const val GROUP_REMOVE_MEMBER = "GROUP_REMOVE_MEMBER"
    const val SEND_GROUP_MSG = "SEND_GROUP_MSG"
    const val LIST_MY_GROUPS = "LIST_MY_GROUPS"
    const val GET_GROUP_INFO = "GET_GROUP_INFO"
    const val SEARCH_GROUPS = "SEARCH_GROUPS"
    const val GROUP_EVENT = "GROUP_EVENT"
    const val GROUP_MSG = "GROUP_MSG"

    // ===== 文件传输 =====
    const val FILE_CHUNK = "FILE_CHUNK"

    // 默认端口
    const val DEFAULT_TCP_PORT = 7891
    const val DEFAULT_DHT_PORT = 7892
    const val DEFAULT_UDP_PORT = 7893

    // 安全参数
    const val PBKDF2_ITERATIONS = 200_000
    const val SALT_SIZE = 16
    const val NONCE_SIZE = 12
    const val KEY_SIZE = 32
    const val TAG_SIZE = 16

    // 心跳间隔（毫秒）
    const val HEARTBEAT_INTERVAL_MS = 30_000L

    // 死人开关默认值
    const val DEFAULT_DEADMAN_GRACE_DAYS = 7
}
